test/Spec.hs

```diff
@@ -60,12 +60,12 @@ main = do
     tests <- goldenTests progress
     defaultMain tests
 
-bUILD_TYPE :: String
---bUILD_TYPE = "debug"
-bUILD_TYPE = "release"
+list :: String -> [String]
+list x = [x]
 
-cargo_build_flag :: [String]
-cargo_build_flag = if bUILD_TYPE == "release" then ["--release"] else []
+cargo_build_flags :: IO [[Char]]
+cargo_build_flags = do
+  (maybe [] (list)) <$> lookupEnv "STACK_CARGO_FLAGS"
```

Comment:
Try replacing `list` -> `return` (cause lists are monads in Haskell)

Reply:
Very nice, thank you!